api/v1alpha1/api.go

```diff
@@ -144,3 +150,120 @@ const (
 	// This can be used to describe the routing behavior in HTTPRoute referenced by LLMRoute.
 	LLMModelHeaderKey = "x-envoy-ai-gateway-llm-model"
 )
+
+// BackendSecurityPolicyType specifies the type of auth mechanism used to access a backend.
+type BackendSecurityPolicyType string
+
+const (
+	BackendSecurityPolicyTypeAPIKey BackendSecurityPolicyType = "APIKey"
+	BackendSecurityPolicyTypeAWSIAM BackendSecurityPolicyType = "AWS_IAM"
```

Comment:
can we have a consistent style?
```suggestion
	BackendSecurityPolicyTypeAPIKey BackendSecurityPolicyType = "APIKey"
	BackendSecurityPolicyTypeAWSIAM BackendSecurityPolicyType = "AWSIAM"
```

ack AWSIAM looks confusing though ... @arkodg @zhaohuabing how do you guys usually name types like this

Reply:
ha they are both acryonyms, so accoding to https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#constants it should be `AWSIAM`, but I agree this doesnt read well

Reply:
yeah let's leave it as-is!